test/e2ev2/tmanager/manager.go

```diff
@@ -119,6 +153,41 @@ func (tm *TestManagerIbc) Start() {
 	tm.UpdateWalletsAccSeqNumber()
 }
 
+// Start runs all the nodes, PreUpgradeFunc and processes upgrade to the latest tag
+func (tm *TestManagerUpgrade) Start(govMsg *govtypes.MsgSubmitProposal, preUpgradeFunc PreUpgradeFunc) {
```

Comment:
let's not couple the start and the upgrade together, TestManagerUpgrade could have an Upgrade function that just sends the proposal and votes from every validator on it 


What do you think?